### ISSUE TITLE
`requestLanguageModelAccess` API doesn't lead me to be defensive

### ISSUE THREAD

**Issue by TylerLeonhardt** [microsoft/vscode#205723]
Testing #205609

Originally I had this line:

```ts
const access = await lm.requestLanguageModelAccess(GPT_35);
```

where `GPT_35` was: `copilot-gpt-3.5`.

But what I discovered was that `copilot-gpt-3.5` was not actually a model.. it has `-turbo` on the end. So I find myself doing:
```ts
        let access;
        try {
            access = await lm.requestLanguageModelAccess(GPT_35);
        } catch(e) {
            access = await lm.requestLanguageModelAccess(lm.languageModels[0]);
        }
```

So that my extension can work in case we change the name again. This isn't really an obvious path forward and not to mention it is error prone if `lm.languageModels[0]` isn't a similar model.

I almost want to say: "give me one of these models, but if not, I'll take anything from Copilot." And make that clear in API instead of having to do nasty try/catches or go through `lm.languageModels` looking for any that start with `copilot`

**Comment by jrieken** [comment]
> And make that clear in API instead of having to do nasty try/catches or go through 

Maybe I am missing something but afaik you will always need try-catch because a model cannot exist or access hasn't been granted.

**Comment by TylerLeonhardt** [comment]
Yes that's true. I guess my meta question is, how will we prevent stale extensions from breaking if the model doesn't exist in the future.

**Comment by jrieken** [comment]
We have removed the access object indirection. It's now just `vscode.lm.sendChatRequest` which returns a promise that rejects when anything happens that prevents the request from being made.
